### PR TITLE
pybind/rados: add more type hintings and cleanup

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/rados/rados.pyx [hunk 1]
```diff
@@ -249,7 +249,7 @@ cdef extern from "rados/librados.h" nogil:
     void rados_ioctx_snap_set_read(rados_ioctx_t io, rados_snap_t snap)
     int rados_ioctx_snap_list(rados_ioctx_t io, rados_snap_t * snaps, int maxlen)
     int rados_ioctx_snap_get_stamp(rados_ioctx_t io, rados_snap_t id, time_t * t)
-    uint64_t rados_ioctx_get_id(rados_ioctx_t io)
```
Comment (tchaikov) [inline src/pybind/rados/rados.pyx:252]: @jdurgin this is the change i mentioned the other day when discussing the API doc build.

**Review by jdurgin — APPROVED** [review]

**Review by neha-ojha — APPROVED** [review]

**Comment by tchaikov** [comment]
changelog

- fix the annotation of the return type of `get_locator_key()`. was `bytes`, should be `str`.

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-10-13_01:10:32-rados-wip-kefu-testing-2020-10-12-1615-distro-basic-smithi
- https://pulpito.ceph.com/kchai-2020-10-14_11:01:53-rados-wip-kefu-testing-2020-10-14-1541-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-10-15_08:32:19-rados-wip-kefu-testing-2020-10-14-1541-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-10-15_10:09:42-rados-wip-kefu-testing-2020-10-14-1541-distro-basic-smithi/

**Pull request merged**